app/controllers/admin/greet_controller.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+class Admin::GreetController < AdminApplicationController
+  before_action :authenticate_user!, :ensure_privileges!
```

Comment:
You can use only ensure_privileges method and do the check there for the authenticate.

Reply:
Makes more sense this way indeed. Will update.